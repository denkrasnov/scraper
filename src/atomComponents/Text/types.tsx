import { ReactNode } from "react";

type FontWeight = 200 | 400 | 700;

type FontSize =
  | "fs100" // 10px, step: 2
  | "fs200"
  | "fs300"
  | "fs400"
  | "fs500"
  | "fs600"
  | "fs700"
  | "fs800"
  | "fs1000";

export type Sizes = { [key in FontSize]: string };

type Tag = "p" | "span";

type TextAlign = "center";

export interface TextProps {
  children: ReactNode;
  fontWeight?: FontWeight;
  fontSize?: FontSize;
  tag?: Tag;
  textAlign?: TextAlign;
}
