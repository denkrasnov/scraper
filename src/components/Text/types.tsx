import { ReactNode } from "react";

export type FontWeight = "fw200" | "fw400" | "fw700";

export type FontSize =
  | "fs100" // 10px, step: 2
  | "fs200"
  | "fs300"
  | "fs400"
  | "fs500"
  | "fs600"
  | "fs700"
  | "fs800"
  | "fs1000";

export type Sizes = { [key in FontSize]: string };
export type Weights = { [key in FontWeight]: number };

type Tag = "p" | "span";

type TextAlign = "center";

export interface TextProps {
  children: ReactNode;
  fontWeight?: FontWeight;
  fontSize?: FontSize;
  tag?: Tag;
  textAlign?: TextAlign;
}
