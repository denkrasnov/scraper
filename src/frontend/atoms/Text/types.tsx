import { ReactNode } from "react";

import { Color } from "../Colors/types";

export type FontWeight = "fw200" | "fw400" | "fw700";

export type FontSize =
  | "fs10"
  | "fs12"
  | "fs14"
  | "fs16"
  | "fs18"
  | "fs20"
  | "fs22"
  | "fs24"
  | "fs26";

type Tag = "p" | "span";

type TextAlign = "center";

export interface TextProps {
  children: ReactNode;
  color?: Color;
  fontWeight?: FontWeight;
  fontSize?: FontSize;
  tag?: Tag;
  textAlign?: TextAlign;
}
