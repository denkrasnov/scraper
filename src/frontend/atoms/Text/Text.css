@import "../Colors/color.css";

.textAlign-center {
  text-align: center;
}

.fontSize-fs10 {
  font-size: 10px;
}

.fontSize-fs12 {
  font-size: 12px;
}

.fontSize-fs14 {
  font-size: 14px;
}

.fontSize-fs16 {
  font-size: 16px;
}

.fontSize-fs18 {
  font-size: 18px;
}

.fontSize-fs20 {
  font-size: 20px;
}

.fontSize-fs24 {
  font-size: 24px;
}

.fontSize-fs26 {
  font-size: 26px;
}

.fontWeight-fw200 {
  font-weight: 200;
}

.fontWeight-fw400 {
  font-weight: 400;
}

.fontWeight-fw700 {
  font-weight: 700;
}

@media (min-width: 992px) {
  .gradientHover {
    position: relative;
    display: inline-block;
    overflow: hidden;
    background: linear-gradient(
      to right,
      var(--body-text),
      var(--body-text) 50%,
      var(--text-gray) 50%
    );
    background-clip: text;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    background-size: 200% 100%;
    background-position: 100%;
    transition: background-position 275ms ease;
    text-decoration: none;
  }

  .gradientHover:hover {
    background-position: 0 100%;
  }
}
