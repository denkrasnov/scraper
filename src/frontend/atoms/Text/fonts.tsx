import { Sizes, Weights } from "./types";

export const sizes: Sizes = {
  fs100: "10px",
  fs200: "12px",
  fs300: "14px",
  fs400: "16px",
  fs500: "18px",
  fs600: "20px",
  fs700: "22px",
  fs800: "24px",
  fs1000: "26px"
};

export const weights: Weights = {
  fw200: 200,
  fw400: 400,
  fw700: 700
};
