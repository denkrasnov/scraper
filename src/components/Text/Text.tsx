import React, { FC } from "react";

import { sizes, weights } from "./fonts";
import { TextProps, Sizes, Weights } from "./types";

const Text: FC<TextProps> = props => {
  const { children, tag, fontSize, fontWeight, ...rest } = props;
  const Element = tag || "span";

  const style: { [key: string]: string | number } = {};

  if (fontSize) style.fontSize = sizes[fontSize];
  if (fontWeight) style.fontWeight = weights[fontWeight];

  return <Element style={{ ...style, ...rest }}>{children}</Element>;
};

export default Text;
