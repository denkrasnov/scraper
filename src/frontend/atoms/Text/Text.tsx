import React, { FC } from "react";

import colors from "../Colors";
import { StyleType } from "../helpers/types";
import { sizes, weights } from "./fonts";
import { TextProps } from "./types";

const Text: FC<TextProps> = props => {
  const { children, color, tag, fontSize, fontWeight, ...rest } = props;
  const Element = tag || "span";

  const style: StyleType = {};

  if (fontSize) style.fontSize = sizes[fontSize];
  if (fontWeight) style.fontWeight = weights[fontWeight];
  if (color) style.color = colors[color];

  return <Element style={{ ...style, ...rest }}>{children}</Element>;
};

export default Text;
