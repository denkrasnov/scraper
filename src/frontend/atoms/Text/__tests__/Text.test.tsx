import React from "react";
import { shallow } from "enzyme";

import Text from "..";

describe("Text", () => {
  const children = "__TEXT__";

  it("should render", () => {
    const component = shallow(<Text>{children}</Text>);

    expect(component).toMatchSnapshot();
  });

  it("should render with attributes", () => {
    const component = shallow(
      <Text color="BODY_TEXT" fontSize="fs10" fontWeight="fw400">
        {children}
      </Text>
    );

    expect(component).toMatchSnapshot();
  });
});
