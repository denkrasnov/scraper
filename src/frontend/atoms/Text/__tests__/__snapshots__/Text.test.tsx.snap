// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render 1`] = `
<span
  className="root"
>
  __TEXT__
</span>
`;

exports[`Text should render with attributes 1`] = `
<span
  className="root color-BODY_TEXT fontSize-fs10 fontWeight-fw400"
>
  __TEXT__
</span>
`;
